### PR TITLE
Add overview summary for backups

### PR DESCRIPTION
**Pull request by piitaya**
## Proposed change

Add overview summary card at the top
Move fetching and subscribe logic to backup root component.
Update icons

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/backup/ha-config-backup-settings.ts [hunk 1]
```diff
@@ -211,16 +174,17 @@ class HaConfigBackupSettings extends LitElement {
...
+        include_addons: this.config!.create_backup.include_addons ?? [],
+        include_all_addons: this.config!.create_backup.include_all_addons,
+        password: this.config!.create_backup.password,
       },
-      retention: this._backupConfig.retention,
-      schedule: this._backupConfig.schedule.state,
+      retention: this.config!.retention,
+      schedule: this.config!.schedule.state,
     });
+    fireEvent(this, "ha-refresh-backup-config");
```
Comment (bramkragten) [inline src/panels/config/backup/ha-config-backup-settings.ts:187]: Maybe we should only do this on disconnected? As it will cause a lot of fetches?

File: src/panels/config/backup/ha-config-backup-settings.ts [hunk 2]
```diff
@@ -196,10 +159,10 @@ class HaConfigBackupSettings extends LitElement {
 
   private _encryptionKeyChanged(ev) {
     const password = ev.detail.value as string;
-    this._backupConfig = {
-      ...this._backupConfig,
+    this.config = {
```
Comment (bramkragten) [inline src/panels/config/backup/ha-config-backup-settings.ts:162]: I don't think we should change a public property, it could be overwritten by the parent before it is saved.

File: src/panels/config/backup/ha-config-backup-settings.ts (hunk 1 above)
Comment (piitaya) [reply src/panels/config/backup/ha-config-backup-settings.ts:187]: Yeah agree 🙂

File: src/panels/config/backup/ha-config-backup-settings.ts (hunk 2 above)
Comment (piitaya) [reply src/panels/config/backup/ha-config-backup-settings.ts:162]: It was a state before, I forgot to change this logic

File: src/panels/config/backup/ha-config-backup-settings.ts (hunk 1 above)
Comment (piitaya) [reply src/panels/config/backup/ha-config-backup-settings.ts]: It can not be done in disconnected because it's already disconnected so the element will be fired.
Also, I don't think people will change these settings too often. 
Should I try to improve it in another way or is it ok like that ?

Comment (bramkragten) [reply src/panels/config/backup/ha-config-backup-settings.ts]: It is ok for now 👍🏻 would be nice if the update call would return the new config like other APIs do

File: src/panels/config/backup/ha-config-backup-settings.ts [hunk 3]
```diff
@@ -51,22 +27,24 @@ class HaConfigBackupSettings extends LitElement {
...
+      this._config = this.config;
     }
   }
 
-  private async _fetchData() {
-    const { config } = await fetchBackupConfig(this.hass);
-    this._backupConfig = config;
+  protected firstUpdated(_changedProperties: PropertyValues): void {
+    super.firstUpdated(_changedProperties);
+    fireEvent(this, "ha-refresh-backup-config");
```
Comment (bramkragten) [inline src/panels/config/backup/ha-config-backup-settings.ts:43]: Why is this needed?

File: src/panels/config/backup/ha-config-backup-settings.ts [hunk 4]
```diff
@@ -51,22 +27,24 @@ class HaConfigBackupSettings extends LitElement {
...
+  @state() private _config?: BackupConfig;
 
-  protected willUpdate(changedProps) {
-    super.willUpdate(changedProps);
-    if (!this.hasUpdated) {
-      this._fetchData();
+  protected willUpdate(changedProperties: PropertyValues): void {
+    super.willUpdate(changedProperties);
+    if (changedProperties.has("config")) {
+      this._config = this.config;
```
Comment (bramkragten) [inline src/panels/config/backup/ha-config-backup-settings.ts:37]: Shouldn't it only do this if `_config` is undefined?

**Review by bramkragten — APPROVED** [review]
